components/stock-operation-modal/index.js

```diff
@@ -9,45 +13,103 @@ const StockOperationModal = (props) => {
     showModal,
     transactionType,
     stockId,
+    availableQty,
   } = props;
 
   const [quantity, setQuantity] = useState('');
+  const [isUserLoggedIn, setIsUserLoggedIn] = useState('');
+  const [userMoney, setUserMoney] = useState();
+
+  const validateQuantity = (quantity) => {
+    if (quantity > availableQty) {
+      alert(`You can only ${transactionType} ${availableQty} stocks`);
+      setQuantity(availableQty);
+    } else {
+      setQuantity(quantity);
+    }
+  };
 
   const closeModal = () => {
     showModal((prev) => !prev);
     setQuantity('');
   };
-  const submitHandler = () => {
-    const body = {
-      tradeType: transactionType,
-      stockName: nameOfStock,
-      stockId,
-      quantity,
-      listedPrice: listedPriceOfStock,
-      totalPrice: quantity * listedPriceOfStock,
-    };
 
-    fetch('https://api.realdevsquad.com/trade/stock/new/self', {
+  useEffect(() => {
+    getUserWallet();
+  }, []);
+
+  const getUserWallet = async () => {
+    const response = await fetchData(`${BASE_API_URL}/wallet`, 'GET', {
       credentials: 'include',
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json',
-      },
-      body: JSON.stringify(body),
-    })
-      .then((response) => {
-        if (response.status === 200) {
-          return response.json();
-        } else {
-          throw new Error(response.statusText);
+    });
+    const { wallet } = await response.json();
+    if (Object.keys(wallet).length === 0) return setUserMoney(0);
+    else {
+      const {
+        currencies: { dinero },
+      } = wallet;
+      setUserMoney(dinero);
+    }
```

Comment:
The `else` condition can be removed here, since line no 46 has a `return` statement.

Reply:
okay! Thanks for the review. 😃

Reply:
Yep :)